helix-tui/src/backend/crossterm.rs

```diff
@@ -23,13 +23,42 @@ use std::{
     fmt,
     io::{self, Write},
 };
+use termini::TermInfo;
 
 fn term_program() -> Option<String> {
-    std::env::var("TERM_PROGRAM").ok()
+    // Some terminals don't set $TERM_PROGRAM
+    match std::env::var("TERM_PROGRAM") {
+        Err(_) => std::env::var("TERM").ok(),
+        Ok(term_program) => Some(term_program),
+    }
 }
 fn vte_version() -> Option<usize> {
     std::env::var("VTE_VERSION").ok()?.parse().ok()
 }
+fn reset_cursor_approach(terminfo: TermInfo, fallback: &str) -> String {
```

Comment:
```suggestion
fn reset_cursor_escape_sequence(terminfo: TermInfo) -> String {
```
 Passing the fallback as an argument is a bit qeord/unnecessary just put it into the function

Reply:
sorry about that, it is a habit of my make all state come into the function through the arguments. I've undone it.